components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -304,6 +316,42 @@ pub fn cast_uint_as_string(val: &Option<Int>) -> Result<Option<Bytes>> {
     }
 }
 
+/// The implementation for push down signature `CastIntAsJson` from unsigned integer.
+#[rpn_fn]
+#[inline]
+pub fn cast_uint_as_json(val: &Option<Int>) -> Result<Option<Json>> {
+    match val {
+        None => Ok(None),
+        Some(val) => Ok(Some(Json::U64(*val as u64))),
+    }
+}
+
+#[rpn_fn(capture = [extra])]
+#[inline]
+pub fn cast_string_as_json(
+    extra: &RpnFnCallExtra<'_>,
+    val: &Option<Bytes>,
+) -> Result<Option<Json>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            if extra
+                .ret_field_type
+                .flag()
+                .contains(FieldTypeFlag::PARSE_TO_JSON)
+            {
+                let s = box_try!(String::from_utf8(val.to_owned()));
+                let val: Json = s.parse()?;
+                Ok(Some(val))
+            } else {
+                // FIXME: port `JSONBinary` from TiDB to adapt if the bytes is not a valid utf8 string
+                let val = unsafe { String::from_utf8_unchecked(val.to_owned()) };
```

Comment:
Prefer to implement `ConvertTo<Json> for Bytes` and then call this convert, so that it becomes unified with the rest of the convert function.

Reply:
This function depends on `FieldType`, so cannot use `ConvertTo<Json>` to unify with others.